recipe/build.sh

```diff
@@ -39,6 +39,27 @@ for path in $SLEPC_DIR $BUILD_PREFIX; do
     done
 done
 
+# Patch some linking variables post-configure
+# PETSC_SNES_LIB contains _all_ petsc external libraries
+# resulting in major over-linking
+# further, SLEPC_LIB also pulls in the full PETSC_SNES_LIB, when it should be just petsc, slepc,
+# and SLEPC_LIB_BASIC is also incorrect
+# also note that we have to use -lscalapack, not ${SCALAPACK_LIB}
+# because slepc config explicitly forces an empty `SCALAPACK_LIB = `
+
+slepcvariables=$(ls installed-*/lib/slepc/conf/slepcvariables)
+echo "patching $slepcvariables"
+test -f $slepcvariables
+
+cat >> $slepcvariables << 'EOF'
+# conda-forge overrides to avoid over-linking
+SLEPC_LIB_BASIC = -lslepc
+PETSC_SNES_LIB = ${CC_LINKER_SLFLAG}${SLEPC_LIB_DIR} -L${SLEPC_LIB_DIR} ${PETSC_LIB_BASIC} -lblas -llapack -lscalapack
```

Comment:
Could you try the following one?

```suggestion
PETSC_SNES_LIB = ${CC_LINKER_SLFLAG}${SLEPC_LIB_DIR} -L${SLEPC_LIB_DIR} ${PETSC_LIB_BASIC} ${BLASLAPACK_LIB} ${SCALAPACK_LIB}
```

Reply:
This make the thing robust upon changes on the BLAS/LAPACK implementation. Some day conda-force may consider moving to flexiblas.

Reply:
And sorry for being such a PITA!

Reply:
No problem. I actually did that first, but went with this because slepcvariables explicitly clears `SCALAPACK_LIB = ` first, so I get an empty string. So I'd have to specify SCALAPACK_LIB to the correct value and then use it, which seems to miss the point.

After this, I trust manual, explicit library names a lot more than petsc and slepc's internal variables, which seem to be frequently wrong and very hard to debug (SCALAPACK_LIB, SLEPC_LIB_BASIC, and all SLEPC_*_LIB variables have the wrong values in slepcvariables).

Reply:
> slepcvariables explicitly clears `SCALAPACK_LIB = ` 

This can be considered a bug in SLEPc. I will fix.

Reply:
https://gitlab.com/slepc/slepc/-/merge_requests/627

Reply:
Got it, thank you! I thought maybe it was the right thing to do internally, since it knew it was getting scalapack via petsc, based on the log output and the fact that it does work.